src/appshell/qml/Window.qml

```diff
@@ -28,8 +28,30 @@ DockWindow {
     property var provider: InteractiveProvider {
         topParent: dockWindow
         onRequestedDockPage: {
+            initParams(uri, params)
             dockWindow.currentPageUri = uri
         }
+
+        function initParams(uri, params) {
+            if (!Boolean(params)) {
+                return
+            }
+
+            for (var i in dockWindow.pages) {
+                var page = dockWindow.pages[i]
+                if (page.uri === uri) {
```

Comment:
You can simplify this method:

for (var i in dockWindow.pages) {
var page = dockWindow.pages[i]

if (page.uri !== uri) {
continue
}
...
}